tests/unittest_inference.py

```diff
@@ -2051,6 +2051,30 @@ def test_dict_invalid_args(self) -> None:
             self.assertIsInstance(inferred, Instance)
             self.assertEqual(inferred.qname(), "builtins.dict")
 
+    def test_copy_method_inference(self) -> None:
+        code = """
+        a = {"b": 1, "c": 2}
+        b = a.copy()
+        b #@
+
+        a = [1, 2, 3]
+        b = a.copy()
+        b #@
+
+        a = set([1, 2, 3])
+        b = a.copy()
+        b #@
+
+        a = frozenset([1, 2, 3])
+        b = a.copy()
+        b #@
```

Comment:
I would suggest to choose different variable names for each case. That could make debugging a bit easier if something were to go wrong at some point.

Could you also add a case for which we **can't** infer `copy()`?